src/main/java/MinesweeperCell.java

```diff
@@ -10,7 +10,8 @@
     public boolean wasClicked;
     public int value; // value = MINE or 0-8: nrOfAdjacent mines
     public static final int MINE = -1;
-
+    public boolean isRevealed;
+    public boolean isFlagged;
```

Comment:
You don't need booleans for this.
You can just create methods to check to see what the Image is.